### PR TITLE
gh-527 Ensure that the feature toggle works for the Dashboard

### PR DESCRIPTION
**Pull request by ghillert**
* Add feature toggle
  - uses the `/about` REST endpoint
  - treats feature toggle like an authorization concern
  - uses `RolesDirective` to make checks
* Add tests

resolves https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/210
resolves https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/527

### CODE REVIEW

File: ui/src/app/shared/model/about/feature-info.model.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        case 'tasksEnabled': {
+          return this.tasksEnabled;
+        }
+        default: {
+          console.error(`Unsupported feature parameter '${feature}'.`);
+          return false;
+        }
+      }
+    } else {
+      return false;
```
Comment (cppwfs) [inline ui/src/app/shared/model/about/feature-info.model.ts]: may also want to have a console out here, just to report that feature wasn't present.

File: ui/src/app/shared/services/shared-about.service.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { Http, Response, RequestOptionsArgs } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/merge';
+import 'rxjs/add/observable/of';
+
+import { AppRegistration, ApplicationType, DetailedAppRegistration, ErrorHandler, Page } from '../model';
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Can be removed not used.

File: ui/src/app/shared/services/shared-about.service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { PageRequest } from '../model/pagination/page-request.model';
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Can be removed not used.

File: ui/src/app/shared/services/shared-about.service.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { HttpUtils } from '../support/http.utils';
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Can be removed not used.

File: ui/src/app/shared/services/shared-about.service.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { FeatureInfo } from '../model/about/feature-info.model';
+import { Subject } from 'rxjs/Subject';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
```
Comment (cppwfs) [inline ui/src/app/shared/services/shared-about.service.ts]: Can be removed not used.

File: ui/src/app/auth/logout.component.ts [hunk 6]
```diff
@@ -41,10 +43,10 @@ export class LogoutComponent implements OnInit {
     }
 
     console.log('Logging out ...');
-
     if (this.authService.securityInfo.isFormLogin) {
       this.authService.logout().subscribe(
         result => {
+          this.aboutService.featureInfo.reset();
```
[the remark this answers is not in the document]
Comment (ghillert) [reply ui/src/app/auth/logout.component.ts]: @cppwfs I don't think it matters that much as a failure would be somewhat unexpected. Nonetheless, will add the reset :-)

**Comment by cppwfs** [comment]
After latest update I'm picking up test failures
* AboutDetailsComponent: Should display base information and services are enabled

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Rebased Squashed Merged

**Pull request closed without merging**